src/msg/async/rdma/RDMAConnectedSocketImpl.cc

```diff
@@ -603,7 +603,7 @@ void RDMAConnectedSocketImpl::notify()
   int ret;
 
   ret = write(notify_fd, &i, sizeof(i));
-  assert(ret = sizeof(i));
+  assert(ret == sizeof(i));
```

Comment:
@joscollin 
write() returns number of bytes that were written That means Its not TYPE, its value.
sizeof() returns size(in bytes) of its operand or type passed. Not type itself.

So I suppose this comparison is type-safe as both Left, Right hand returns ints not types
assert(sizeof(notify_fd) == write(notify_fd, &i, sizeof(i)));